packages/amplify-category-function/src/provider-utils/awscloudformation/utils/functionPluginLoader.ts

```diff
@@ -238,3 +246,7 @@ interface ListOption {
   name: string;
   value: string;
 }
+
+function isLayerParameter(params: Partial<LayerParameters> | Partial<FunctionParameters> ): params is Partial<LayerParameters> {
+  return (((params as Partial<LayerParameters>) !== undefined) && ((params as Partial<LayerParameters>).runtimes !== undefined));
```

Comment:
nit don't need the type assertion to check for undefined. Also, I would say this is a typeguard for `Pick<LayerParameters, 'runtimes'>` rather than `Partial<LayerParameters>`

Reply:
If I am using params.runtime , its giving error as runtimes is not part of functionParameters.
Thats  why I put a typeguard to differentiate between layer and function parameters.

Is there any other way to do this?

Reply:
You need the typeguard for the second check, just not the first one